### DISCUSSION TITLE
Help with DocOnce setup4book needed.

### DISCUSSION THREAD

**Discussion by przemarbor** [doconce/doconce#230] [Q&A] [answered]
Hi,
I'm interested in using DocOnce for writing my documents. I decided to start from DocOnce Book stencil authored by HP Langtangen in order to avoid starting from the scratch:
http://hplgit.github.io/setup4book-doconce/doc/web/index.html
https://github.com/hplgit/setup4book-doconce
http://hplgit.github.io/setup4book-doconce/doc/pub/book.pdf

I'm working on Debian Buster and installed doconce via Debian repository and publish via pip.
I hoped that after cloning setup4book repository everything would work out of the box. Unfortunately, I keep getting the errors while trying to compile this document. 
I know that I can't count on getting an answer from the original author, that's why I would like to ask you for the help.

Here's the bash script with the commands I run to test the repository: 
https://pastebin.com/nx6THiPJ

And here are the log files with the errors after each command:
logFile1.log: https://pastebin.com/BVfdePyj
logFile2.log: https://pastebin.com/QvMv8ULh
logFile3.log: https://pastebin.com/Zr58nEiB
logFile4.log: https://pastebin.com/hEcEyPpz

I guess I must be missing something obvious but I can't figure out what it is...
I'd really appreciate your hints.

**Comment by lexnederbragt** [comment, marked as answer]
Thanks for your interest in using DocOnce. First off, he setup4book-doconce repository is no longer maintained, but that doesn't meant it isn't a good place to start. What I don't know is whether you'll be able to get the whole thing to work, though - but let's take it one step at a time.

From your log files, at least numbers 1-3, it seems you are missing the `publish` package. You can install this via `publish-doconce`. It is used to handle references. Let me know if installing this helps and solves the problems in your first three files, and what happens (if anything) to the problem reported in logFile4.

**Reply by przemarbor** [reply]
Thanks for your reply. After some experiments with the code I deduced that in fact something was wrong with publish package. After commenting line: 
`# BIBFILE: ../papers.pub`
all the compilation process goes like a charm (of course with no references in the final document).
Yet, as I mentioned in the previous post I did install publish package and now when I try to run:
    `pip3 install publish-doconce`
system replies:
`Requirement already satisfied: publish-doconce in /home/mb/.local/lib/python3.7/site-packages (1.1.4)`
`Requirement already satisfied: lxml in /home/mb/.local/lib/python3.7/site-packages (from publish-doconce) (4.6.2)`
`Requirement already satisfied: python-Levenshtein-wheels in /home/mb/.local/lib/python3.7/site-packages (from publish-doconce) (0.13.2)`

However, if I try to run publish on some BiBTeX file:
`>> publish import papers.bib`
I obtain:
`Traceback (most recent call last):`
`File "/home/mb/.local/bin/publish", line 6, in <module>`
    `from publish import main`
`ImportError: cannot import name 'main' from 'publish' (/home/mb/.local/lib/python3.7/site-packages/publish/__init__.py)`

Shall I reinstall publish-doconce? Have I installed in incorrectly?

**EDIT**
I can fix the problem by working with virtual environments:
`python3 -m venv venv`
`source venv/bin/activate`
`pip3 install -r requirements.txt`

where the content of `requirements.txt` was generated during my first attempts with doconce and publish and is uploaded [here](https://github.com/przemarbor/zobib2pubib/blob/main/requirements.txt). 

**EDIT 2**
Ha! I forgot about `sudo`!
`sudo pip3 install doconce-publish`
solves the problem.

**Reply by lexnederbragt** [reply]
Good to hear, because I would not have known how to solve this for you :-). Good luck with working with DocOnce and don't hesitate to continue discussing it here! Closing this now
